### PR TITLE
Thing's configurations now posting to '/things/{UID}/config'

### PR DESCRIPTION
**Pull request by aounhaider1**
1) Things configurations are now updated through /things/{UID}/config rest endpoint as described here https://github.com/eclipse/smarthome/issues/771
2) Also fixes the issue https://github.com/eclipse/smarthome/issues/960

Signed-off-by: Aoun Bukhari <bukhari@itemis.de>

### CODE REVIEW

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js [hunk 1]
```diff
@@ -505,7 +505,7 @@ angular.module('PaperUI.controllers.configuration',
 		} else {
 		    thing.item = {};
         }
-		thingSetupService.update(thing, function() {
+		thingService.updateConfig({thingUID:thing.UID},thing.configuration, function() {
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js]: There is no save action for the Eclipse JS addon, but if you use "Source, Format" manually I would assume that it adds spaces after a comma.

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js [hunk 2]
```diff
@@ -476,7 +476,7 @@ angular.module('PaperUI.controllers.configuration',
         $mdDialog.hide(itemName);
     }
 }).controller('EditThingController', function($scope, $mdDialog, toastService, 
-		thingTypeRepository, thingRepository, thingSetupService, homeGroupRepository, configService) {
+		thingTypeRepository, thingRepository, thingSetupService, homeGroupRepository, configService,thingService) {
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js]: see formatter comment...

**Comment by maggu2810** [comment]
1. Thanks
2. LGTM, just two minor remarks

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js (hunk 1 above)
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js]: Okay, there are save actions for JS but it seems to be not working... Don't know why.

Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js]: Yes I tried them, even with different profiles but they doesn't seem to work. So normally I just use the keyboard shortcut.

**Pull request merged**